include/sentry.h

```diff
@@ -1386,6 +1416,15 @@ SENTRY_EXPERIMENTAL_API void sentry_set_span(sentry_transaction_t *tx);
  * finishing the Span means it will be discarded, and will not be sent to
  * sentry. `sentry_value_null` will be returned if the child Span could not be
  * created.
+ *
+ * The Transaction passed in must have a read lock to be thread-safe. This
```

Comment:
"read lock" again somewhat odd, since it suggest some kind of RwLock but that's not true since this does mutation.  i do think you mean "mutex" in these cases.